File: client/src/model/ArchiveMark.ts
import Department from './Department'

interface IArchiveMark {
	code: string
	name: string
	department: Department
}

class ArchiveMark {
	code: string
	name: string
	department: Department

	constructor(obj?: IArchiveMark) {
		this.code = (obj && obj.code) || ''
		this.name = (obj && obj.name) || ''
		this.department = (obj && obj.department) || null
	}
}

export default ArchiveMark

File: server/DocumentsKM/Models/ArchiveMark.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentsKM.Models
{
    // Марка
    public class ArchiveMark
    {
        // Название
        [MaxLength(255)]
        public string Name { get; set; }

        // Код
        [Required]
        [MaxLength(30)]
        public string Code { get; set; }

        // Отдел
        [Required]
        public int DepartmentId { get; set; }

        public Department Department { get; set; }
    }
}
